lib/Stan_math_backend.ml

```diff
@@ -2,77 +2,59 @@
 
 open Core_kernel
 open Mir
-open Format
+open Fmt
 
-let comma ppf () = fprintf ppf ",@ "
-let newline ppf () = fprintf ppf "@;"
-let semi_new ppf () = fprintf ppf "@;"
-let emit_str ppf s = fprintf ppf "%s" s
 let zero = Lit (Int, "0")
 
-let emit_option ppf (emitter, opt, default) =
-  match opt with
-  | Some x -> fprintf ppf "%a" emitter x
-  | None -> emit_str ppf default
-
-let rec emit_unsizedtype ad ppf = function
-  | Ast.UInt | UReal -> emit_str ppf ad
-  | UArray t -> fprintf ppf "std::vector<%a>" (emit_unsizedtype ad) t
-  | UMatrix -> fprintf ppf "Eigen::Matrix<%s, -1, -1>" ad
-  | URowVector -> fprintf ppf "Eigen::Matrix<%s, 1, -1>" ad
-  | UVector -> fprintf ppf "Eigen::Matrix<%s, -1, 1>" ad
+let rec pp_unsizedtype ad ppf = function
```

Comment:
The convention in Fmt is to create "pretty printers" for your type named `pp_<typename>`: https://erratique.ch/software/fmt/doc/Fmt.html#nameconv

Reply:
OK! Would it make sense to change the filename to `Code_generation` then? Or something similar? Not a big deal either way.

Reply:
I'm not sure I understand... It's called Stan_math_backend rather than the more generic `Code_generation` because we will probably have multiple code generation backends. Right now we just have the C++ Stan Math backend, but we want to keep it easy to support Tensorflow Probability as a backend as well.

Reply:
Isn't `backend` a bit more general? In my mind, the backend of a compiler does more than just code generation. It may also do various low-level optimizations.

How about `Stan_math_code_gen`?

Reply:
Sounds good!